### ISSUE TITLE
[4.x] $value can be instance of Collection

### ISSUE THREAD

**Issue by Anubarak** [engram-design/SuperTable#555]
### Describe the bug

The `$value` return value from `$element->getFieldValue('superTableField')` can be an instance of `\craft\elements\Collection` instead of `craft\db\elements\ElementQuery` 

https://github.com/verbb/super-table/blob/craft-4/src/fields/SuperTableField.php#L868

```PHP
/** @var SuperTableBlockQuery|Collection $value */
$value = $element->getFieldValue($this->handle);
// ... snip
$value->setCachedResult($blocks);
```
your IDE should complain about this line as `setCachedResult` does not exist in a `Collection`.
Could you please change this to
```PHP
if (!$allBlocksValidate && $value instanceof ElementQuery) {
    // Just in case the blocks weren't already cached
    $value->setCachedResult($blocks);
}
```

### Steps to reproduce

1.  Eager Load super table field value with invalid blocks due to changed validation rules
2.  see the error `Method craft\\elements\\ElementCollection::setCachedResult does not exist`

### Craft CMS version

4.X

### Plugin version

3.0.12

### Multi-site?

No

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/super-table:"dev-craft-4 as 3.0.12"`.

**Comment by Anubarak** [comment]
Thank you very much 🙇

**Comment by engram-design** [comment]
Fixed in [3.0.13](https://github.com/verbb/super-table/releases/tag/3.0.13)
